### PR TITLE
Remove controller specs

### LINKED ISSUE #1827: Remove rails-controller-testing gem
**Is your feature request related to a problem? Please describe.**

The [rails-controller-testing](https://github.com/rails/rails-controller-testing) gem is supposed to bring back two methods that have been deprecated in Rails: `assigns` and `assert_template` since when Rails started discouraging testing controller internals and views from the controller tests.

This gem though doesn't seem to be used by the app (I grepped for `assigns`, `assert_template` and `Rails::Controller::Testing`) and nothing popped up

**Describe the solution you'd like**

I suggest that it should be removed.

**Describe alternatives you've considered**

Well, no alternatives in this case, unless there's some other tool that's using it

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description
- Moves controller specs to request spec.
- Remove rails-controller-testing gem.
- Reorganize article request specs.
- Add additional specs.
## Related Tickets & Documents
Resolves #1827 
## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
n/a
## Added to documentation?
- [x] no documentation needed

### CODE REVIEW

File: Guardfile [hunk 1]
```diff
@@ -65,6 +65,7 @@ guard :rspec, rspec_options do
     [
       rspec.spec.call("routing/#{m[1]}_routing"),
       rspec.spec.call("controllers/#{m[1]}_controller"),
```
Comment (lightalloy) [inline Guardfile]: Probably, this line is not relevant anymore?

Comment (maestromac) [reply Guardfile]: Oh yeah, thanks!

File: app/models/user.rb [hunk 2]
```diff
@@ -268,7 +268,7 @@ def processed_website_url
   end
 
   def remember_me
-    true
+    Rails.env.test? ? false : true
```
Comment (rhymes) [inline app/models/user.rb]: Maybe you can mock this method in the spec... to avoid having "test logic" in the actual app

Comment (maestromac) [reply app/models/user.rb]: Yeah, let me do that.

Comment (maestromac) [reply app/models/user.rb]: Actually I thought you were referring to something else but I went ahead and removed that. 😆

**Pull request merged**